modules/entity/src/entity_state.ts

```diff
@@ -9,11 +9,14 @@ export function getInitialEntityState<V>(): EntityState<V> {
 
 export function createInitialStateFactory<V>() {
   function getInitialState(): EntityState<V>;
+  function getInitialState<S extends EntityState<V>>(
+    additionalState: Partial<S>
```

Comment:
why do you use Partial here?

Reply:
I mean we should not allow user declare ids or entities, so we should use conditional type here to ban those props.

Reply:
My personal preference is to not any properties that aren't on the extended state, and that includes not allowing ids or entities. That's how my original fix worked. However, since the original typings allowed anything, and the single test confirmed that was the desired behavior, I changed it to a partial without removing the `EntityState<T>` properties. If you'd prefer that we only strictly allow the additional properties explicitly declared on a custom state object, then great - I'd love to restrict it to that as I originally had done! However, that would require deleting the previous test and breaking current functionality.

I can do either, so which do you prefer?

Reply:
This is what I had before, but the existing test suggested it was too restrictive:
```typescript
    getInitialState<S extends EntityState<T>>(
        state: Pick<S, Exclude<keyof S, keyof EntityState<T>>>
    ): S {
        return Object.assign(this.getInitialEntityState(), state) as S;
    }
```
which gives:
```typescript
export const initialState = adapter.getInitialState<StateWithAdditionalProps>({
    // only intellisense for additional properties beyond those in EntityState<T>:
    isHydrated: false,

    // and anything else gives red squigglies, including ids and entities:
    // ids: [],
    // entities: {},
    // foos: [],
});
```

Reply:
That `Exclude` is new in TypeScript 2.8. If we need to use something earlier, then this `Diff` implementation works instead:
```typescript
export type Diff<T extends string, U extends string> = (
    & { [P in T]: P }
    & { [P in U]: never }
    & { [x: string]: never }
)[T];
```
So we just replace `Exclude` with `Diff`.

Reply:
Sorry let me think again. there is a use case that user would like to provide initial entities different from {}.